app-nordigen/bank-factory.js

```diff
@@ -0,0 +1,19 @@
+const IngPlIngbplpw = require('./banks/ing-pl-ingbplpw');
+const IntegrationBank = require('./banks/integration-bank');
+const MbankRetailBrexplpw = require('./banks/mbank-retail-brexplpw');
+const SandboxfinanceSfin0000 = require('./banks/sandboxfinance-sfin0000');
+
+const BankFactory = (institutionId) => {
```

Comment:
Could this be represented as an array instead?

```js
const banks = [MbankRetailBrexplpw, SandboxfinanceSfin0000, IngPlIngbplpw]
// then:
const Bank = banks.find(b => b.insitutionId === id) || IntegrationBank
new Bank()
```

On another level, does this API even need classes? It seems to me like the API could be defined as a static interface with the appropriate keys. If you defined the interface type in a `.ts` file then it could be imported in a JSDoc comment:

```js
/** @type {import('./nordigen-types').Bank} */
module.exports = {
  institutionId: 'foo',
  normalizeAccount(account) {
    // ...
  },
  // ...
}
```

Reply:
Looks very interesting! I will take a look at it! Thanks for your input!